src/manifests/distribution.py

```diff
@@ -0,0 +1,26 @@
+import logging
+import urllib.request
+from typing import List
+
+
+class DistributionNotFound(Exception):
+    def __init__(self, urls: List[str]):
+        self.urls = urls
+        super().__init__(f"Unable to find a distribution under urls {self.urls}")
+
+
+def find_build_root(base_url: str, platform: str, architecture: str, product_name: str) -> str:
+    possible_urls = [
+        f"{base_url}/{platform}/{architecture}/builds/{product_name}",
+        f"{base_url}/{platform}/{architecture}/builds"
+    ]
+
+    for distribution_url in possible_urls:
+        manifest_url = f"{distribution_url}/manifest.yml"
+        try:
+            with urllib.request.urlopen(manifest_url):
+                # OK we could access the manifest, return the url
+                return distribution_url
+        except:
```

Comment:
Check that the exception is a 403 or 404, fail on everything else.

Reply:
Lets fast follow on these so the build isn't blocked, https://github.com/opensearch-project/opensearch-build/issues/1289